client/app/app_config.go

```diff
@@ -100,6 +103,7 @@ var (
 	beginBlockers = []string{
 		minttypes.ModuleName,
 		distrtypes.ModuleName, // Note: slashing happens after distr.BeginBlocker
+		epochstypes.ModuleName,
```

Comment:
The epochs module must come before other modules relying on the epoch system so that they have the latest epoch data.

More precisely, in our context, the epochs module only needs to come before the staking module (that we will introduce later as a custom forked module from Cosmos SDK). I think the distribution and mint modules can be before the epoch module since they do not use epoch data.

Reply:
in this case, why don't we make epoch the first one? just in case we may need to use it for mint/dist later.

Reply:
We should. See [dYdX](https://github.com/dydxprotocol/v4-chain/blob/085609527f3f62a88e7bbc52211a6e12727526f0/protocol/app/app.go#L1299-L1302) and [Osmosis](https://github.com/osmosis-labs/osmosis/blob/0a768877d76a859ffe73e3a0b6716377f8724e37/app/modules.go#L218-L224). But also, alternatively, Babylon puts it after Cosmos SDK modules and before custom modules that use epochs data (see [here](https://github.com/babylonchain/babylon/blob/6dbcf44b79511873aa3840de0b3c1dc8d1b78372/app/app.go#L352-L363)).

Reply:
Thank you @jdubpark for explanation instead of me. 

As jdub said, it just should be front of `evmstaking` module for now since the only module which uses the epoch data is `evmstaking` module. I think it would be better to put at first as @LeoHChen said just in case.